packages/core/src/interpreter.ts

```diff
@@ -695,7 +660,15 @@ export class Actor<TLogic extends AnyActorLogic>
    */
   public getPersistedSnapshot(): Snapshot<unknown>;
   public getPersistedSnapshot(options?: unknown): Snapshot<unknown> {
-    return this.logic.getPersistedSnapshot(this._snapshot, options);
+    const persisted = this.logic.getPersistedSnapshot(this._snapshot, options);
+    if (!this._parent) {
+      const rootPersistedSnapshot = {
+        ...persisted,
+        'xstate.system': this.system.getPersistedSnapshot()
+      };
+      return rootPersistedSnapshot;
+    }
```

Comment:
I'd temporarily disable this part of the code to figure out system-wide rehydration of actor references.

Reply:
This part is removed